command.py

```diff
@@ -0,0 +1,310 @@
+import traceback
+from argparse import ArgumentParser
+
+import extractor
+import logger
+from chunks import Chunks
+from googleservice import GoogleService
+from progresslogger import ProgressLogger, Progress
+from util import guess_mimetype, describe_files, find_last_modified_file
+
+
+class Command:
+    """Commands handle their own argument parser and use GoogleService to execute operations"""
+    TYPE = None
+    HELP = None
+
+    def __init__(self, args):
+        self.args = args
+        self.google = GoogleService()
+
+    @staticmethod
+    def add_to_subparser(subparsers):
+        """Configures ArgParser for the command"""
+        pass
+
+    async def execute(self):
+        pass
+
+    def is_service_started(self):
+        return self.google.is_valid()
+
+
+class Download(Command):
+    TYPE = "download"
+    HELP = "Download a file from Google Drive through ID or Filename."
+
+    def __init__(self, args):
+        super().__init__(args)
+        self.file = " ".join(self.args.file)
+
+    @staticmethod
+    def add_to_subparser(subparsers):
+        parser = subparsers.add_parser(
+            Download.TYPE,
+            help=Download.HELP
+        )
+        parser.set_defaults(command=Download)
+        parser.add_argument(
+            'file',
+            metavar='FILE',
+            nargs='*',
+            help="Name or ID of the file to Download. If not specified, will try as ID then as Name."
+        )
+        name_or_id = parser.add_mutually_exclusive_group()
+        name_or_id.add_argument(
+            '-n',
+            '--name',
+            help='Will use FILE as a filename to search',
+            action='store_true'
+        )
+        name_or_id.add_argument(
+            '-i',
+            '--id',
+            help='Will use FILE as a file ID to download',
+            action='store_true'
+        )
+        parser.add_argument(
+            '-l',
+            '--last',
+            help="Downloads the last modified file",
+            action='store_true'
+        )
+        parser.add_argument(
+            '-e',
+            '--extract',
+            help='Tries to extract the downloaded file',
+            action='store_true'
+        )
+
+    async def execute(self):
+        extract = self.args.extract
+        if self.args.last:
+            await self.download_last_uploaded_file(extract)
+        elif not self.args.file:
```

Comment:
This is not a comment that is part of you change, but just to track it:

I think we could show the help for download here. I mean, if the user does `gdrive download` with no args, I think we could show the `gdrive download --help`

Reply:
Agreed